temporal/api/cloud/connectivityrule/v1/message.proto

```diff
@@ -0,0 +1,53 @@
+syntax = "proto3";
+
+package temporal.api.cloud.connectivityrule.v1;
+
+option go_package = "go.temporal.io/api/cloud/connectivityrule/v1;connectivityrule";
+option java_package = "io.temporal.api.cloud.connectivityrule.v1";
+option java_multiple_files = true;
+option java_outer_classname = "MessageProto";
+option ruby_package = "Temporalio::Api::Cloud::ConnectivityRule::V1";
+option csharp_namespace = "Temporalio.Api.Cloud.ConnectivityRule.V1";
+
+import "temporal/api/cloud/resource/v1/message.proto";
+import "google/protobuf/timestamp.proto";
+
+message ConnectivityRule {
+  // The id of the private connectivity rule.
+  string id = 1;
+
+  // The connectivity rule specification.
+  ConnectivityRuleSpec spec = 2;
+
+  // The current version of the connectivity rule specification.
+  // The next update operation will have to include this version.
+  string resource_version = 3;
+  temporal.api.cloud.resource.v1.ResourceState state = 4;
+  // The id of the async operation that is creating/updating/deleting the connectivity rule, if any.
+  string async_operation_id = 5;
+  // The date and time when the connectivity rule was created.
+  google.protobuf.Timestamp created_time = 6;
+}
+
+// The connectivity rule specification passed in on create/update operations.
+message ConnectivityRuleSpec {
+  oneof connection_type {
+    PublicConnectivityRule public_rule = 1;
+    PrivateConnectivityRule private_rule = 2;
+  }
+}
+
+message PublicConnectivityRule {}
+
+message PrivateConnectivityRule {
+  // Connection id provided by user so we enforce the private connectivity.  This is required both by AWS and GCP
+  string connection_id = 1;
+
+  // For GCP private connectivity service, GCP needs both GCP project id and the Private Service Connect Connection IDs
+  // AWS only needs the connection_id
+  string gcp_project_id = 2;
+
+  // The region of the connectivity rule. 
+  // Example: "aws-us-west-2"
+  string region = 3;
```

Comment:
Do we restrict a namespace to use a rule from another region? 
Can we add comments on what to expect here.